sylvia/tests/ui/method_signature/reply.rs

```diff
@@ -0,0 +1,82 @@
+#![allow(unused_imports)]
+use sylvia::cw_std::{Binary, Response, StdResult};
+use sylvia::types::{InstantiateCtx, ReplyCtx};
+
+pub mod mismatched_params {
+    use super::*;
+
+    pub struct Contract {}
+
+    #[sylvia::contract]
+    impl Contract {
+        pub const fn new() -> Self {
+            Self {}
+        }
+
+        #[sv::msg(instantiate)]
+        fn instantiate(&self, _ctx: InstantiateCtx) -> StdResult<Response> {
+            Ok(Response::new())
+        }
+
+        #[sv::msg(reply, handlers=[on_instantiated], reply_on=success)]
+        fn first_reply(
+            &self,
+            _ctx: ReplyCtx,
+            _data: Option<Binary>,
+            param: String,
+        ) -> StdResult<Response> {
+            Ok(Response::new())
+        }
+
+        #[sv::msg(reply, handlers=[on_instantiated], reply_on=failure)]
+        fn first_reply(
+            &self,
+            _ctx: ReplyCtx,
+            _data: Option<Binary>,
+            param: u32,
+        ) -> StdResult<Response> {
+            Ok(Response::new())
+        }
+    }
+}
+
+pub mod mismatched_param_lenght {
+    use super::*;
+
+    pub struct Contract {}
+
+    #[sylvia::contract]
+    impl Contract {
+        pub const fn new() -> Self {
+            Self {}
+        }
+
+        #[sv::msg(instantiate)]
+        fn instantiate(&self, _ctx: InstantiateCtx) -> StdResult<Response> {
+            Ok(Response::new())
+        }
+
+        #[sv::msg(reply, handlers=[on_instantiated], reply_on=success)]
+        fn first_reply(
```

Comment:
Is it OK to have two methods with the same name in the `impl` block?

Reply:
It is not, thanks for pointing this out.